src/databricks/labs/ucx/hive_metastore/migration_status.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+import logging
+from dataclasses import dataclass
+from collections.abc import Iterable
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MigrationStatus:
+    src_schema: str
+    src_table: str
+    dst_catalog: str | None = None
+    dst_schema: str | None = None
+    dst_table: str | None = None
+    update_ts: str | None = None
+
+    def destination(self):
+        return f"{self.dst_catalog}.{self.dst_schema}.{self.dst_table}".lower()
+
+
+@dataclass
+class TableView:
+    catalog: str
+    schema: str
+    name: str
+
+    def key(self):
+        return f"{self.catalog}.{self.schema}.{self.name}".lower()
+
+
+class MigrationIndex:
+    def __init__(self, tables: list[MigrationStatus]):
+        self._index = {(ms.src_schema, ms.src_table): ms for ms in tables}
+
+    def is_migrated(self, schema: str, table: str) -> bool:
+        """Check if a table is migrated."""
+        return self.get(schema, table) is not None
+
+    def get(self, schema: str, table: str) -> MigrationStatus | None:
```

Comment:
You could go fully pythonic with `__getitem__` instead of `get`

Reply:
Overruled by @nfx